Start updating github landing page and contributing.md.

rishig opened the pull request:
This process will have a lot of tiny decisions, so pushing it out bit by bit.

timabbott commented:
I merged 2 commits.  For the meetings, I think maybe we should just move those to a separate doc to shorten the main doc for the server a bit more, not remove that content.

For "unused mailing lists", I think we should instead rename `zulip-devel` to `zulip-devel-announce` and still leave it documented as lower-priority.  I imagine we should still use it for things like announcing our PyCon gatherings (and GCI).  I'm happy to merge the changes to kill the iOS/Android lists, though.

rishig commented:
Updated. Possible followups include
* Renaming zulip-devel (though not sure zulip-devel-announce is the right name for it). I updated the description on google groups at least to clarify that it is an announce-only list.
* Fleshing out "And beyond" section for advice after your first pull request is merged
* Linking to the git tutorial once we have it
* Removing all or most of the License text at the bottom of README.md

rishig commented:
License updated.

timabbott commented on LICENSE:
```diff
@@ -175,28 +196,3 @@
-   distributed under the License is distributed on an "AS IS" BASIS,
-   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-   See the License for the specific language governing permissions and
-   limitations under the License.
```
Don't remove this.  The instructions for how to apply the license are part of the license text.

timabbott commented on README.md:
```diff
@@ -24,276 +15,63 @@ project.
-
-
-[czo-doc]: https://zulip.readthedocs.io/en/latest/chat-zulip-org.html
+## Next steps
```
This heading feels a bit off, but I can't think of something better.  Maybe add to the list of follow-ups "start a thread about what this should be".

timabbott commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,291 @@
+  important thing to say is whether you're looking for a backend (Python),
+  frontend (JavaScript), mobile (React Native), desktop (Electron),
+  documentation (English) or visual design (JavaScript + CSS) issue, and a
+  bit about your programming experience.
```
Maybe add "and available time"?  I think that would often be helpful.

timabbott commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,291 @@
+Other notes:
+* For a first pull request, it's better to aim for a smaller contribution
+  than a bigger one. Many first contributions have fewer than 10 lines of
+  changes (not counting tests).
```
Maybe "(not counting changes to tests)"

timabbott commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,291 @@
+
+* Posting good questions. This generally means explaining your current
+  understanding, saying what you've done or tried so far, and including
+  tracebacks or other error messages if appropriate.
```
Maybe include a recommendation to use code blocks rather than screenshots?  Probably just a thing to add a link to a doc on screenshots and screencasts; feel free to put this in a follow-up issue.

timabbott commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,291 @@
+## Reporting issues
+
+If you find an easily reproducible bug and/or are experienced in reporting
+bugs, feel free to just open an issue on the relevant project on GitHub.
```
We have a ReadTheDocs article on reporting bugs; we should link to it.  `docs/bug-reports.md`.  It's not the best; the good content in there is the links.

timabbott commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,291 @@
+[Google Summer of Code](https://developers.google.com/open-source/gsoc/),
+and the
+[MIT Externship program](https://alum.mit.edu/students/NetworkwithAlumni/ExternshipProgram),
+and also takes summer interns from Harvard, MIT, and Stanford.
```
s/also takes/has in the past taken/.  I don't want to make it sound like a requirement.

timabbott commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,291 @@
+
+## What makes a great Zulip contributor?
+
+Zulip runs a lot of internship programs, so we have a lot of experience with
```
make "internship programs" a link.

timabbott commented:
I posted a few comments.  I think we're ready to merge after fixing these (and then we'll want to open follow-up issues for some of the TODO items).

rishig commented:
Resolved most of them. For the rest:
* post as a code block not a screenshot: not sure how to work this in, but it's there in chat-zulip-org.md. Opened #7388 to give something to link to in the future. 
* Link to docs/bugs-report.md: I think the doc needs work, opened #7388 to give something to link to in the future.

Remaining followup work is 
* Rename Next Steps (though I think it is fine for now)
* Fleshing out "And beyond" section for advice after your first pull request is merged
* Linking to the git tutorial once we have it

(we decided not to bother renaming zulip-devel)

timabbott commented:
Great, merged, thanks @rishig!  

@rosatamsen FYI, since this is probably going to cause a few merge conflicts with your work (I think most should be easy to resolve)
